src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs

```diff
@@ -21,34 +25,90 @@ public static TItem Get<TItem>(this IMemoryCache cache, object key)
 
         public static bool TryGetValue<TItem>(this IMemoryCache cache, object key, out TItem value)
         {
-            object obj = null;
-            if (cache.TryGetValue(key, out obj))
+            object result;
+            if (cache.TryGetValue(key, out result))
             {
-                value = (TItem)obj;
+                value = (TItem)result;
                 return true;
             }
+
             value = default(TItem);
             return false;
         }
 
-        public static object Set(this IMemoryCache cache, object key, object value)
+        public static TItem Set<TItem>(this IMemoryCache cache, object key, TItem value)
         {
-            return cache.Set(key, value, new MemoryCacheEntryOptions());
+            return cache.Set(key, value, null);
         }
 
-        public static object Set(this IMemoryCache cache, object key, object value, MemoryCacheEntryOptions options)
+        public static TItem Set<TItem>(this IMemoryCache cache, object key, TItem value, MemoryCacheEntryOptions options)
         {
-            return cache.Set(key, value, options);
+            using (var entry = cache.CreateEntry(key))
+            {
+                if (options != null)
+                {
+                    if (options.AbsoluteExpiration.HasValue)
```

Comment:
MemoryCacheEntryOptions and ICacheEntry need to be reconciled. The two should work mostly the same.